idea-plugin/src/main/kotlin/arrow/meta/ide/plugins/quotes/resolve/QuoteSyntheticPackageFragmentProvider.kt

```diff
@@ -19,53 +20,51 @@ import org.jetbrains.kotlin.resolve.scopes.DescriptorKindFilter
 import org.jetbrains.kotlin.resolve.scopes.MemberScope
 import org.jetbrains.kotlin.utils.Printer
 
-val IdeMetaPlugin.metaSyntheticPackageFragmentProvider: ExtensionPhase
+val IdeMetaPlugin.quoteSyntheticPackageFragmentProvider: ExtensionPhase
   get() = packageFragmentProvider { project, module, _, _, _, _ ->
     descriptorCachePackageFragmentProvider(module, project)
   }
 
 private val descriptorCachePackageFragmentProvider: (ModuleDescriptor, Project) -> PackageFragmentProvider
   get() = { module, project ->
-    val cache = QuoteSystemCache.getInstance(project)
+    val quoteCache: QuoteCache? = project.getService(QuoteCache::class.java)
```

Comment:
I think that accepting an unavailable service here may introduce bugs, which are hard to spot. If `null`, then resolving may silently fail. As far as I can see there should always be a `QuoteCache` in the project.

Reply:
Yes, before it was ill typed. And that silent fail was implied. This will silently fail, and let other SyntheticPackageFragmentProvider proceed